hicommonwealth/commonwealth
# Add Bounties

## Description

- Add Bounties Controller to Substrate shared files
  - `bounty` as an extension of the proposal type,
  - `bountyTreasury` is a similar structure to treasury controller, but handles bounties.
- Add Bounty page and view under proposals (rendering bounty like a proposal)

TODO: 
- [x] Fix compilation issue with ApiPromise (@polkadot/api versioning problem)
- [x] Create Bounty Page

Closes #828.

## Motivation and Context




## How to test



To get it running:
- Go to `/chain-events` repo, branch `zak.bountyce` and `yarn build`. 
- Run `yarn unlink` and `yarn link` in the repo to prepare local package sharing.
- Go to `/commonwealth` repo, branch `zak.bounties`, run `yarn link @commonwealth/chain-events`. 
- Run `yarn` and then `yarn start`. 
- Create bounties on a local-edgeware chain: `target/release/edgeware --chain=local --pruning=archive --alice --force-authoring`

QA UI:
- Go to Edgeware-local,
- Go to Bounties page, view bounties, click on a bounty, see bounty details.

## Have proper tags been added (for bug, enhancement, breaking change)?

## Does this PR affect any server routes?
